dask/array/lib/stride_tricks.py

```diff
@@ -0,0 +1,71 @@
+from ..overlap import rechunk_for_overlap, map_overlap, trim_chunks
+
+
+def _window_view(arr, window_shape, axis, fix_first_block, block_info):
+    """ core sliding window function"""
+    from ..numpy_compat import sliding_window_view
+
+    block_info = block_info[0]
+    slicer = [slice(None, None)] * arr.ndim
+    for ax, window, fix_first in zip(axis, window_shape, fix_first_block):
+        start, stop = None, None
+        is_last_chunk = (
+            block_info["num-chunks"][ax] - 1 - block_info["chunk-location"][ax]
+        ) == 0
+
+        locs = block_info["array-location"][ax]
+        size = locs[1] - locs[0]
+        if not is_last_chunk:
+            stop = -1
+
+        # when depth == chunk size of first chunk along some axis
+        # we need to make some confusing empirical adjustments.
+        # could be avoided if we rechunked so that min
+        # chunk size is depth+1 (which would also be more efficient)
```

Comment:
All this indexing stuff has a strong code smell to it. Using `block_info` adds overhead, so it would be nice if it could be eliminated. Can you explain the logic?